File: LeetCode (C++)/81-search-in-rotated-sorted-array-ii.cpp
class Solution {
public:
    bool search(vector<int>& nums, int target) {
        int start = 0;
        int end = nums.size() - 1;
        while (true) {
            if (nums[start] == target or nums[end] == target) {
                return true;
            }
            start += 1;
            end -= 1;
            if (end < start) {
                return false;
            }
        }
    }
};

File: LeetCode (Python 3)/81-search-in-rotated-sorted-array-ii.py
class Solution:
    def search(self, nums: List[int], target: int) -> bool:
        start = 0
        end = len(nums) - 1
        while True:
            if nums[start] == target or nums[end] == target:
                return True
            start += 1
            end -= 1
            if end < start:
                return False